### PR TITLE
aggregate.d, replace `for` loops with `foreach` over aggregates

### PR DESCRIPTION
**Pull request by ghost**
Also addresses a few D-Scanner warnings for the module
- line len over 120
- local var can be const

(https://github.com/dlang/dmd/projects/27)

### CODE REVIEW

File: src/dmd/aggregate.d [hunk 1]
```diff
@@ -193,7 +193,7 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             if (v.aliassym)
                 return 0;   // If this variable was really a tuple, skip it.
 
-            if (v.storage_class & (STC.static_ | STC.extern_ | STC.tls | STC.gshared | STC.manifest | STC.ctfe | STC.templateparameter))
+            with (STC) if (v.storage_class & (static_ | extern_ | tls | gshared | manifest | ctfe | templateparameter))
```
Comment (jacob-carlborg) [inline src/dmd/aggregate.d]: I have a suspicion Walter won't like this.

Comment (BBasile) [reply src/dmd/aggregate.d]: Okay, cancelled this change.

**Review by wilzbach — APPROVED** [review]

**Comment by WalterBright** [comment]
Be extremely careful about converting loops in dmd to foreach. The reason is that some of the routines can add members to the arrays while in the loop. A number of these changes had to be reverted because of that.

Basically, do not do this conversion unless you are absolutely sure that isn't happening.

File: src/dmd/aggregate.d [hunk 2]
```diff
@@ -730,11 +730,8 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
                 }
             }
 
-            for (size_t i = 0; i < members.dim; i++)
-            {
-                auto sm = (*members)[i];
+            foreach (sm; *members)
```
Comment (WalterBright) [inline src/dmd/aggregate.d]: this loop can add members, do not convert

File: src/dmd/aggregate.d [hunk 3]
```diff
@@ -216,9 +216,8 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             return 0;
         }
 
-        for (size_t i = 0; i < members.dim; i++)
+        foreach (s; *members)
```
Comment (WalterBright) [inline src/dmd/aggregate.d]: This loop can add members, please revert

**Pull request merged**